django/db/backends/oracle/introspection.py

```diff
@@ -106,52 +106,29 @@ def get_table_list(self, cursor):
         ]
 
     def get_table_description(self, cursor, table_name):
-        """
-        Return a description of the table with the DB-API cursor.description
-        interface.
-        """
-        # user_tab_columns gives data default for columns
+        # First, we get default_collation in a separate query
         cursor.execute(
             """
-            SELECT
-                user_tab_cols.column_name,
-                user_tab_cols.data_default,
-                CASE
-                    WHEN user_tab_cols.collation = user_tables.default_collation
-                    THEN NULL
-                    ELSE user_tab_cols.collation
-                END collation,
-                CASE
-                    WHEN user_tab_cols.char_used IS NULL
-                    THEN user_tab_cols.data_length
-                    ELSE user_tab_cols.char_length
-                END as display_size,
-                CASE
-                    WHEN user_tab_cols.identity_column = 'YES' THEN 1
-                    ELSE 0
-                END as is_autofield,
-                CASE
-                    WHEN EXISTS (
-                        SELECT  1
-                        FROM user_json_columns
-                        WHERE
-                            user_json_columns.table_name = user_tab_cols.table_name AND
-                            user_json_columns.column_name = user_tab_cols.column_name
-                    )
-                    THEN 1
-                    ELSE 0
-                END as is_json,
-                user_col_comments.comments as col_comment
-            FROM user_tab_cols
-            LEFT OUTER JOIN
-                user_tables ON user_tables.table_name = user_tab_cols.table_name
-            LEFT OUTER JOIN
-                user_col_comments ON
-                user_col_comments.column_name = user_tab_cols.column_name AND
-                user_col_comments.table_name = user_tab_cols.table_name
-            WHERE user_tab_cols.table_name = UPPER(%s)
-            """,
-            [table_name],
+            SELECT default_collation
+            FROM user_tables
+            WHERE table_name = %s
+        """,
+            [table_name.upper()],
+        )
+        row = cursor.fetchone()
+        default_collation = row[0] if row else None
+
+        # Then we pass default_collation to the main query
+        cursor.execute(
+            """
+            SELECT column_name, data_type, data_default, nullable,
+                data_length, data_precision, data_scale, char_used,
+                char_length
+            FROM user_tab_columns
+            WHERE table_name = %s
+            ORDER BY column_id
```

Comment:
Why you removed all the logic for other columns? (`is_json`, `is_autofield`, etc.) It seems that you don't have an Oracle setup.